client/src/rpc/client.rs

```diff
@@ -372,6 +375,13 @@ impl RpcClient {
                 }),
         )
     }
+
+    /// Updates session enclaves.
+    pub fn update_enclaves(&self, enclaves: Option<HashSet<EnclaveIdentity>>) -> () {
+        let mut session = self.inner.session.lock().unwrap();
+        session.builder = session.builder.clone().remote_enclaves(enclaves);
```

Comment:
Maybe it should check if the enclave identities differ from the existing ones and do nothing in case they don't.